pkg/executor/push_test.go

```diff
@@ -252,74 +252,99 @@ func TestImageNameTagDigestFile(t *testing.T) {
+		{description: "localhost registry with config", Destinations: []string{"localhost:5000/test-image"}, ExistingConfig: true, CheckPushPermsExpectedCallCount: 1},
+		{description: "any other registry", Destinations: []string{"notgcr.io/test-image"}, CheckPushPermsExpectedCallCount: 1},
+		{
+			description: "multiple destinations pushed to different registry",
+			Destinations: []string{
 				"us-central1-docker.pkg.dev/prj/test-image",
 				"us-west-docker.pkg.dev/prj/test-image",
 			},
-			[]bool{true, true}, false,
+			CheckPushPermsExpectedCallCount: 2,
 		},
-		{"same image names with different tags",
-			[]string{
+		{
+			description: "same image names with different tags",
+			Destinations: []string{
 				"us-central1-docker.pkg.dev/prj/test-image:tag1",
 				"us-central1-docker.pkg.dev/prj/test-image:tag2",
 			},
-			[]bool{true, true}, false,
+			CheckPushPermsExpectedCallCount: 1,
 		},
-		{"same destination image multiple times",
-			[]string{
+		{
+			description: "same destination image multiple times",
+			Destinations: []string{
 				"us-central1-docker.pkg.dev/prj/test-image",
 				"us-central1-docker.pkg.dev/prj/test-image",
 			},
-			[]bool{true, false}, false,
+			CheckPushPermsExpectedCallCount: 1,
+		},
+		{
+			description:                     "no push and no push cache",
+			Destinations:                    []string{"us-central1-docker.pkg.dev/prj/test-image"},
+			CheckPushPermsExpectedCallCount: 0,
+			NoPush:                          true,
+			NoPushCache:                     true,
+		},
+		{
+			description:                     "no push and push cache",
+			Destinations:                    []string{"us-central1-docker.pkg.dev/prj/test-image"},
+			CacheRepo:                       "us-central1-docker.pkg.dev/prj/cache-image",
+			CheckPushPermsExpectedCallCount: 1,
+			NoPush:                          true,
+		},
+		{
+			description:                     "no push and cache repo is OCI image layout",
+			Destinations:                    []string{"us-central1-docker.pkg.dev/prj/test-image"},
+			CacheRepo:                       "oci:/some-layout-path",
+			CheckPushPermsExpectedCallCount: 0,
+			NoPush:                          true,
 		},
 	}
 
 	checkRemotePushPermission = fakeCheckPushPermission
 	for _, test := range tests {
 		t.Run(test.description, func(t *testing.T) {
-			setCalledFalse()
+			resetCalledCount()
 			fs = afero.NewMemMapFs()
 			opts := config.KanikoOptions{
-				Destinations: test.Destination,
+				CacheRepo:    test.CacheRepo,
+				Destinations: test.Destinations,
+				NoPush:       test.NoPush,
+				NoPushCache:  test.NoPushCache,
 			}
 			if test.ExistingConfig {
 				afero.WriteFile(fs, util.DockerConfLocation(), []byte(""), os.FileMode(0644))
 				defer fs.Remove(util.DockerConfLocation())
 			}
 			CheckPushPermissions(&opts)
-			for i, shdCall := range test.ShouldCallExecCommand {
-				if i < len(calledExecCommand) && shdCall != calledExecCommand[i] {
```

Comment:
Because `calledExecCommand` is never mutated, it is always 0 length, and this statement is always false; I wasn't exactly sure what this was originally supposed to be testing, but I added `checkPushPermsCallCount` to give some more coverage.